tensorzero-core/src/config/snapshot.rs

```diff
@@ -161,6 +170,39 @@ impl ConfigSnapshot {
         }
     }
 
+    /// Create a ConfigSnapshot from data loaded from the database.
+    ///
+    /// This is used when loading a previously stored config snapshot from ClickHouse.
+    /// The hash is recomputed from the config and templates to ensure consistency.
+    pub fn from_stored(
+        config_toml: &str,
+        extra_templates: HashMap<String, String>,
+        original_hash: &SnapshotHash,
+    ) -> Result<Self, Error> {
+        let table: toml::Table = config_toml.parse().map_err(|e| {
+            Error::new(ErrorDetails::Serialization {
+                message: format!("Failed to parse stored config TOML: {e}"),
+            })
+        })?;
+
+        let sorted_table = prepare_table_for_snapshot(table);
+        let config = UninitializedConfig::try_from(sorted_table.clone())?;
+        let hash = ConfigSnapshot::hash(&sorted_table, &extra_templates)?;
+        if hash != *original_hash {
+            return Err(Error::new(ErrorDetails::ConfigSnapshotHashMismatch {
```

Comment:
Is there a way for us to know how often this happens (maybe via otel)? That would point to some serious issues and I think we should understand it.

Reply:
(non-blocking)